pkg/certificate/manager.go

```diff
@@ -56,6 +47,103 @@ func (m *Manager) Start(checkInterval time.Duration, stop <-chan struct{}) {
 	}()
 }
 
+func (m *Manager) start(ctx context.Context, mrcClient MRCClient) error {
+	// start a watch and we wait until the manager is initialized so that
+	// the caller gets a manager that's ready to be used
+	var once sync.Once
+	var wg sync.WaitGroup
+	mrcEvents, err := mrcClient.Watch(ctx)
+	if err != nil {
+		return err
+	}
+
+	wg.Add(1)
+
+	go func(wg *sync.WaitGroup, once *sync.Once) {
+		for {
+			select {
+			case <-ctx.Done():
+				if ctx.Err() != nil {
+					log.Error().Err(err).Msg("context canceled with error. stopping MRC watch...")
+					return
+				}
+
+				log.Info().Msg("context canceled. stopping MRC watch...")
+				return
+			case event, open := <-mrcEvents:
+				if !open {
+					// channel was closed; return
+					log.Info().Msg("stopping MRC watch...")
+					return
+				}
+
+				err = m.handleMRCEvent(mrcClient, event)
+				if err != nil {
+					log.Error().Err(err).Msgf("error encountered processing MRCEvent")
+					continue
+				}
+			}
+
+			if m.signingIssuer != nil && m.validatingIssuer != nil {
+				once.Do(func() {
+					wg.Done()
+				})
+			}
+		}
+	}(&wg, &once)
+
+	done := make(chan struct{})
+
+	// Wait for WaitGroup to finish and notify select when it does
+	go func() {
```

Comment:
nit: this pattern looks well suited for the sync.Cond type

Reply:
Oh I didn't know go had a cond data structure! Yeah, I'll try to implement this

Reply:
On second thought, this scenario seems a bit too complicated for a cond; we don't have multiple readers or writers. Plus, it's harder to implement a timeout since a cond is based on locks not channels